### PR TITLE
[AIRFLOW-1065] Add functionality for Azure Blob Storage over wasb://

### PR DESCRIPTION
**Pull request by hgrif**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-XXX

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR implements a hook to interface with Azure storage over wasb://
via azure-storage; adds sensors to check for blobs or prefixes; and
adds an operator to transfer a local file to the Blob Storage.

Design is similar to that of the S3Hook in airflow.operators.S3_hook.

Added functionality for Azure Blob Storage:

- `airflow.contrib.hooks.wasb_hook.WasbHook`: Interface with Azure Blob Storage.
- `airflow.contrib.operators.file_to_wasb.FileToWasbOperator`: Upload a file to Azure Blob Storage.
- `airflow.contrib.sensors.wasb_sensor.WasbBlobSensor`: Check if a blob is has landed.
- `airflow.contrib.sensors.wasb_sensor.WasbPrefixSensor`: Check if any blobs matching a prefix have landed.

The design for the hook airflow.contrib.hooks.WasbsHook mimics airflow.operators.S3_hook.S3Hook.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

- `airflow/tests/contrib/hooks/wasb_hook.py`
- `airflow/tests/contrib/operators/file_to_wasb.py`
- `airflow/tests/contrib/sensors/wasb_sensor.py`

Note that the tests mock the `BlockBlobService` of `azure-storage`. For full automated testing the [Azure Storage Emulator](https://docs.microsoft.com/en-us/azure/storage/storage-use-emulator) should be set up.

To test manually: 

- Create a Virtual Machine and Blob Block Storage on Azure.
- Install and set up airflow on the Virtual Machine.
- Test DAGs with a login & key (Storage Account -> Access keys).
- Test DAGs with a login & SAS token (Storage Account -> Shared access signature).

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -155,6 +155,7 @@ def write_version(filename=os.path.join(*['airflow',
 slack = ['slackclient>=1.0.0']
 statsd = ['statsd>=3.0.1, <4.0']
 vertica = ['vertica-python>=0.5.1']
+wasb = ['azure-storage>=0.34.0']
```
Comment (bolkedebruin) [inline setup.py]: Isn't "azure" more intuitive?

File: tests/contrib/hooks/wasb_hook.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
```
Comment (bolkedebruin) [inline tests/contrib/hooks/wasb_hook.py]: Your tests are not executed on Travis. You will need to include them in "init.py" or rename the file to test_XXX.py

File: tests/contrib/operators/file_to_wasb.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
```
Comment (bolkedebruin) [inline tests/contrib/operators/file_to_wasb.py]: Idem

File: tests/contrib/sensors/wasb_sensor.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
```
Comment (bolkedebruin) [inline tests/contrib/sensors/wasb_sensor.py]: Idem

**Review by bolkedebruin — CHANGES_REQUESTED** [review]
Great stuff! Some minor nits

**Comment by hgrif** [comment]
Thanks for your comments!

Looking at the Travis logs, it seems that the unit tests for most other hooks, operators & sensors in `tests/contrib` also don't run (e.g. `datadog_sensor.py`, `emr_base_sensory.py`, but I'm no Travis expert). What would be the best approach here? File a new issue and consistently rename all of them with `test_*`? 

(I like the `test_` approach because `pytest` also uses it.)

**Comment by bolkedebruin** [comment]
Nice! LGTM

**Pull request closed without merging**